### PR TITLE
fix: [CDS-2151]: Add validation on pipeline run for multiple manifests

### PR DESCRIPTION
**Pull request by abosii-harness**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31523)

### CODE REVIEW

**Review by yogesh-chauhan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31523)* status: 5 file(s) to review. 0 of 1 approvals obtained. 

___
*[125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/manifest/ManifestsPlanCreator.java, line 192 at r1](https://reviewable.io/reviews/harness/harness-core/31523#-MyO5Xue5UCWY6VCqE9t:-MyO5Xuf8GA0NR9MtYIX:b5dwq0q) ([raw file](https://github.com/harness/harness-core/blob/2d4be0659c3af04cdf2b988be7d8a66338a15b16/125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/manifest/ManifestsPlanCreator.java#L192)):*
> ```Java
>             manifestIdType, String.join(", ", K8S_SUPPORTED_MANIFEST_TYPES), manifestIdListAsString));
>       }
>     }
> ```

do we need to do this for Helm as well ?

File: 125-cd-nextgen/src/main/java/io/harness/cdng/k8s/utils/K8sStepConstants.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import io.harness.annotations.dev.OwnedBy;
+import io.harness.cdng.manifest.ManifestType;
+
+import com.google.common.collect.ImmutableSet;
+import java.util.Set;
+
+@OwnedBy(CDP)
+public final class K8sStepConstants {
+  public static Set<String> K8S_SUPPORTED_MANIFEST_TYPES = ImmutableSet.of(
+      ManifestType.K8Manifest, ManifestType.HelmChart, ManifestType.Kustomize, ManifestType.OpenshiftTemplate);
```
Comment (shubham-in10se) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/k8s/utils/K8sStepConstants.java:21]: nit: we can move this constant list to `ManifestType` enum itself.

**Review by shubham-in10se — APPROVED** [review]

**Review by abosii-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31523)* status: 7 file(s) to review. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/manifest/ManifestsPlanCreator.java, line 192 at r1](https://reviewable.io/reviews/harness/harness-core/31523#-MyO5Xue5UCWY6VCqE9t:-Mz-XXK77CtpCYVGsR6X:b-6z1kl5) ([raw file](https://github.com/harness/harness-core/blob/2d4be0659c3af04cdf2b988be7d8a66338a15b16/125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/manifest/ManifestsPlanCreator.java#L192)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

do we need to do this for Helm as well ?
</blockquote></details>

Added for Native Helm as well
___
*[125-cd-nextgen/src/main/java/io/harness/cdng/k8s/utils/K8sStepConstants.java, line 21 at r1](https://reviewable.io/reviews/harness/harness-core/31523#-MyIT6iuEQZSvZ-ygpaA-r1-21:-Mz-Xa1E6OmIdeRv8KF4:b-7mmp82) ([raw file](https://github.com/harness/harness-core/blob/2d4be0659c3af04cdf2b988be7d8a66338a15b16/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/utils/K8sStepConstants.java#L21)):*
<details><summary><i>Previously, shubham-in10se (Shubham Maheshwari) wrote…</i></summary><blockquote>

nit: we can move this constant list to `ManifestType` enum itself.
</blockquote></details>

moved

**Review by yogesh-chauhan — DISMISSED** [review]
> Reviewed 7 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31523)* status: Some of the checks are too old. 1. unresolved discussions.

**Review by tchaurasiya — DISMISSED** [review]
> Reviewed 7 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31523)* status: Some of the checks are too old. 1. unresolved discussions.

**Review by acasian-harness — DISMISSED** [review]
> Reviewed 6 of 7 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31523)* status: 1. unresolved discussions.

**Review by yogesh-chauhan — DISMISSED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31523)* status: Some of the checks are too old.

**Review by shubham-in10se — APPROVED** [review]
> Reviewed 6 of 7 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31523)* status: Some of the checks are too old.

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 6 of 6 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31523)* status: :shipit: complete! ✓

**Pull request merged**